sei-db/sc/memiavl/db.go

```diff
@@ -89,6 +93,27 @@ const (
 	SnapshotDirLen = len(SnapshotPrefix) + 20
 )
 
+// getSnapshotModTime returns the modification time of the current snapshot directory.
+// It reads the "current" symlink to get the actual snapshot directory.
+// If the directory doesn't exist or there's an error, returns current time.
+func getSnapshotModTime(dir string) time.Time {
+	// Read the "current" symlink to get the actual snapshot directory
+	currentLink := currentPath(dir)
+	snapshotName, err := os.Readlink(currentLink)
+	if err != nil {
+		// If we can't read the symlink, use current time as fallback
+		return time.Now()
+	}
+
+	snapshotDir := filepath.Join(dir, snapshotName)
+	info, err := os.Stat(snapshotDir)
+	if err != nil {
+		// If we can't get the modification time, use current time as fallback
+		return time.Now()
+	}
+	return info.ModTime()
```

Comment:
Could we think of anything going wrong by relying on OS file system mod time? For example, the mod time might not represent the actual time the snapshot was taken if a validator restores the files from some backup. 

Perhaps, we should consider internal data structure in DB to maintain this type of metadata to avoid a whole bunch of edge cases.